### PR TITLE
Changed to accept a test address to bind

### LINKED ISSUE #10927: In tests, allow changing address to bind to
An instance of `play.test.TestServer` (and therefore also the `WithServer` helper) always binds to `0.0.0.0`. There is no way to change that. I don't like this approach, since even when running tests I don't want to expose my app to the whole world.

We can already change the port a test server is running on by using the system property `testserver.port`:
https://github.com/playframework/playframework/blob/9a1138bce538aa68fdf19ced7ce3b55064aeb62d/testkit/play-test/src/main/scala/play/api/test/Helpers.scala#L138-L142

IMHO we should allow to do the same with the address. I think it would even make sense to allow to set the address by reading a env variable like we already do with `PLAY_HTTP_ADDRESS` (with fallback to the sys property), which let's you set the address in dev and prod mode (however I probably would use `PLAY_TEST_HTTP_ADDRESS` for this case)

### PR DESCRIPTION
**Pull request by morellik**
# Pull Request Checklist

* [ x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [ x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x ] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x ] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [ x] Have you added copyright headers to new files?
* [x ] Have you checked that both Scala and Java APIs are updated?
* [x ] Have you updated the documentation for both Scala and Java sections?
* [x ] Have you added tests for any changed functionality?

## Fixes

Fixes #10927

## Purpose

Allow changing address to bind to

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one commit author (morelli@cerm.unifi.it) is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: dev-mode/run-support/src/test/scala/play/runsupport/FilterArgsSpec.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ class FilterArgsSpec extends Specification {
   val defaultHttpAddress = "0.0.0.0"
 
   def check(args: String*)(
-      properties: Seq[(String, String)] = Seq.empty,
+      properties: Seq[(String, String)] = Seq,
```
Comment (mkurz) [inline dev-mode/run-support/src/test/scala/play/runsupport/FilterArgsSpec.scala:15]: Please undo this change, thanks!

**Review by mkurz — CHANGES_REQUESTED** [review]
Thanks! Please see the other comment I made.

Also if you look at the status of the pull request you see that Travis is not happy. If you look into it, you will see that some files are not correctly formatted:
```
[warn] /home/travis/build/playframework/playframework/testkit/play-test/src/main/scala/play/api/test/Helpers.scala isn't formatted properly!
[warn] /home/travis/build/playframework/playframework/testkit/play-test/src/main/scala/play/api/test/TestServer.scala isn't formatted properly!
[warn] /home/travis/build/playframework/playframework/testkit/play-test/src/main/scala/play/api/test/TestServerFactory.scala isn't formatted properly!
```
(See https://app.travis-ci.com/github/playframework/playframework/jobs/562788374#L317)

You can format them by running `scalafmt` in sbt.

After that, please push your changes again. Thanks!

**Comment by morellik** [comment]
Done

File: testkit/play-test/src/main/scala/play/api/test/Helpers.scala [hunk 2]
```diff
@@ -140,6 +140,7 @@ trait PlayRunners extends HttpVerbs {
    * testserver.port
    */
   lazy val testServerPort: Int = sys.props.get("testserver.port").map(_.toInt).getOrElse(19001)
+  lazy val testServerAddress: String = sys.props.get("testserver.address").orElse(sys.env.get("PLAY_TEST_SERVER_HTTP_ADDRESS")).getOrElse("0.0.0.0")
```
Comment (BillyAutrey) [inline testkit/play-test/src/main/scala/play/api/test/Helpers.scala:143]: If you allow the reference.conf to optionally parse for this environment variable, you don't need the middle part here.

File: dev-mode/run-support/src/test/scala/play/runsupport/FilterArgsSpec.scala (hunk 1 above)
Comment (morellik) [reply dev-mode/run-support/src/test/scala/play/runsupport/FilterArgsSpec.scala:15]: I followed the suggestions of mkurz posted here https://github.com/playframework/playframework/issues/10927

File: testkit/play-test/src/main/scala/play/api/test/Helpers.scala (hunk 2 above)
Comment (morellik) [reply testkit/play-test/src/main/scala/play/api/test/Helpers.scala:143]: I followed the suggestions of mkurz posted here https://github.com/playframework/playframework/issues/10927

**Comment by PromanSEW** [comment]
Should be targeted to `main` branch instead
Also I don't understand why GitHub did not add link issue to "Successfully merging this pull request may close these issues" block...

**Comment by mkurz** [comment]
@morellik thank you very much, however I merged #11509 instead which also contained tests and docs updates.

**Pull request closed without merging**